packages/embedded-provider-core/src/embedded-provider.ts

```diff
@@ -68,6 +74,7 @@ export interface EmbeddedProviderEventMap {
   connect_error: ConnectErrorEventData;
   disconnect: DisconnectEventData;
   error: any;
+  spending_limit_reached: { error: any };
```

Comment:
Now we have a pre-defined error instance, we should use it over `any`.

```suggestion
  spending_limit_reached: { error: SpendingLimitError };
```